sessions/user.go

```diff
@@ -64,15 +70,18 @@ type User struct {
 
 // NewUser Creates a new user session struct object
 func NewUser(conn net.Conn, user *db.User) *User {
-	return &User{
-		Conn:              conn,
-		ConnMutex:         &sync.Mutex{},
-		token:             utils.GenerateRandomString(64),
-		Info:              user,
-		Mutex:             &sync.Mutex{},
-		stats:             map[common.Mode]*db.UserStats{},
-		lastPingTimestamp: time.Now().UnixMilli(),
-		lastPongTimestamp: time.Now().UnixMilli(),
+	sessionUser := User{
+		Conn:                                conn,
+		ConnMutex:                           &sync.Mutex{},
+		PacketChannel:                       make(chan interface{}, 256),
```

Comment:
Have you stress tested this channel? Not sure if a buffer size of 256 is large enough or if having a channel here is a good choice.

Imagine the player is in a 16-player multiplayer game, while spectating, having other users connecting/disconnecting, sending chat messages, and the server pinging them. You can see how this potentially can result in the channel getting full.

Uf the channel ends up getting full, it will block on subsequent sending goroutines until the channel starts being read (which means certain users will be locked).

Reply:
I have increased the size to 1024 based on the fact that when spectating a 16 player game, each sends a packet every 0.4 seconds, meaning 40/s; We allow users to be temporarily disconnected for at most 10 seconds, so at most 400 packets can stack there by spectating only. Considering other packets, this is doubled for safety. You can see [here](https://github.com/Quaver/Z/blob/5ac79ab17bda3f9a0d72a676329146287c76c5ec/sessions/packets.go#L56) that when the channel is full, the packet won't be appended to queue, so it won't block the sending goroutine.

Reply:
Let's say if it does get full while the user is disconnected. They then reconnect. If the packets are no longer appended, then there will be packet loss if they do end up reconnecting. It's not a very scalable solution.

Reply:
Ideally it should be able to account for an infinite amount of packets while the user is disconnected.

Reply:
I found https://github.com/smallnest/chanx which provides an unbounded channel implementation. I'll try if that works.

Reply:
I switched to using chanx. It now won't block input and is unbounded.